src/main/java/org/primeframework/mvc/cors/CORSRequestWorkflow.java

```diff
@@ -55,10 +55,21 @@ public void perform(WorkflowChain workflowChain) throws IOException {
           .withAllowedHTTPMethods(corsConfiguration.allowedMethods)
           .withAllowedOrigins(corsConfiguration.allowedOrigins)
           .withDebugEnabled(corsConfiguration.debug)
-          .withDebugger(debugger)
-          .withExcludedPathPattern(corsConfiguration.excludedPathPattern)
-          .withExposedHeaders(corsConfiguration.exposedHeaders)
-          .withPreflightMaxAge(corsConfiguration.preflightMaxAgeInSeconds);
+          .withDebugger(debugger);
+      if (corsConfiguration.excludedPathPattern != null) {
+        corsFilter.withExcludedPathPattern(corsConfiguration.excludedPathPattern);
+      }
+      else if (corsConfiguration.includedPathPattern != null) {
+        corsFilter.withIncludedPathPattern(corsConfiguration.includedPathPattern);
+      }
+      else if (corsConfiguration.includeUriChecker != null) {
+        corsFilter.withIncludedUriChecker(corsConfiguration.includeUriChecker);
+      }
+      else if (corsConfiguration.excludeUriChecker != null) {
+        corsFilter.withExcludedUriChecker(corsConfiguration.excludeUriChecker);
+      }
+      corsFilter.withExposedHeaders(corsConfiguration.exposedHeaders)
+                .withPreflightMaxAge(corsConfiguration.preflightMaxAgeInSeconds);
```

Comment:
I think you can remove all of these `if` checks and just always call the builder `.with*` method. All of our builder methods should handle `null` correctly.